src/components/AnimateBlock/__tests__/AnimateBlock.test.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+
+import {render, screen} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import AnimateBlock from '../AnimateBlock';
+
+const qaId = 'animate-block';
+
+describe('AnimateBlock', () => {
+    test('render AnimateBlock by default', async () => {
+        render(<AnimateBlock qa={qaId} />);
+
+        const component = screen.getByTestId(qaId);
+        expect(component).toBeInTheDocument();
+    });
+
+    test('add className', () => {
+        const className = 'my-class';
+
+        render(<AnimateBlock qa={qaId} className={className} />);
+        const component = screen.getByTestId(qaId);
+
+        expect(component).toHaveClass(className);
+    });
+
+    test('add className', () => {
+        const className = 'my-class';
+
+        render(<AnimateBlock qa={qaId} className={className} />);
+        const component = screen.getByTestId(qaId);
+
+        expect(component).toHaveClass(className);
+    });
+
+    test('use passed style', () => {
+        const style = {color: 'red'};
+
+        render(<AnimateBlock style={style} qa={qaId} />);
+        const component = screen.getByTestId(qaId);
+
+        expect(component).toHaveStyle(style);
+    });
+
+    test('call onUpdate with checked status', async () => {
```

Comment:
What does it means this test?

Reply:
It detects that onScroll event is fired. Actually, after your comment I thought twice and rewrote this test. Please check it one more time